posthog/temporal/data_imports/pipelines/meta_ads/source.py

```diff
@@ -0,0 +1,430 @@
+        start_date = dt.datetime.strptime(time_range["since"], "%Y-%m-%d")
+        end_date = dt.datetime.strptime(time_range["until"], "%Y-%m-%d")
+
+        current_start = start_date
+        while current_start <= end_date:
+            if current_start.month == 12:
+                current_end = current_start.replace(year=current_start.year + 1, month=1, day=1) - dt.timedelta(days=1)
+            else:
+                current_end = current_start.replace(month=current_start.month + 1, day=1) - dt.timedelta(days=1)
+
+            current_end = min(current_end, end_date)
+
+            monthly_time_range = {
+                "since": current_start.strftime("%Y-%m-%d"),
+                "until": current_end.strftime("%Y-%m-%d"),
+            }
+
+            monthly_params = params.copy()
+            monthly_params["time_range"] = json.dumps(monthly_time_range)
+
+            next_url = url
+
+            while next_url:
+                if next_url == url:
+                    response = requests.get(next_url, params=monthly_params)
+                else:
+                    response = requests.get(next_url)
+
+                if response.status_code != 200:
+                    raise Exception(f"Meta API request failed: {response.status_code} - {response.text}")
+
+                data = response.json()
+                yield data
+
+                paging = data.get("paging", {})
+                next_url = paging.get("next")
+
+            current_start = current_end + dt.timedelta(days=1)
+
+
+def meta_ads_source(
+    config: MetaAdsSourceConfig,
+    team_id: int,
+    should_use_incremental_field: bool = False,
+    db_incremental_field_last_value: typing.Any = None,
+    incremental_field: str | None = None,
+    incremental_field_type: IncrementalFieldType | None = None,
+) -> SourceResponse:
+    """A data warehouse Meta Ads source."""
+    name = NamingConvention().normalize_identifier(config.resource_name)
+    schema = get_schemas()[config.resource_name]
+    is_stats = config.resource_name.endswith("_stats")
+
+    def get_rows() -> collections.abc.Iterator[list[dict]]:
+        integration = get_integration(config, team_id)
+        access_token = integration.access_token
+
+        # Determine date range for incremental sync
+        time_range = None
+
+        if should_use_incremental_field:
+            if incremental_field is None or incremental_field_type is None:
+                raise ValueError("incremental_field and incremental_field_type can't be None")
+
+            if db_incremental_field_last_value is None:
+                last_value: dt.date = dt.date.today() - dt.timedelta(days=META_ADS_MAX_HISTORY_DAYS)
+            else:
+                last_value = db_incremental_field_last_value
+
+            start_date = last_value.strftime("%Y-%m-%d")
+            # Meta Ads API is day based so only import if the day is complete
+            end_date = (dt.date.today() - dt.timedelta(days=1)).strftime("%Y-%m-%d")
```

Comment:
Hmm, partial day imports are fine as long as we continue to get the same day repeatedly until its complete. Giving users the flexibility to sync partial data every hour, for example, instead of forcing them to wait a day isn't great UX